.github/workflows/go.yml

```diff
@@ -183,3 +183,16 @@ jobs:
       run: |
         sudo npm install -g markdownlint-cli
         make markdownlint
+
+  benchmark:
+    name: Go benchmark test
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+      - name: Check-out code
+        uses: actions/checkout@v2
+      - name: Run Go benchmark test
+        run: go test -run '^$' -bench . -benchtime 1s -timeout 10m -cpu 4 -v -benchmem ./pkg/...
```

Comment:
why `1s` instead of `1x`? do you need to run each benchmark more than once? I think `1s` is actually the default for `-benchtime`

Reply:
Yes, `1s` is the default for `-benchtime`, '1x' is more reasonable.

src/testing/benchmark.go
```
func initBenchmarkFlags() {
	matchBenchmarks = flag.String("test.bench", "", "run only benchmarks matching `regexp`")
	benchmarkMemory = flag.Bool("test.benchmem", false, "print memory allocations for benchmarks")
	flag.Var(&benchTime, "test.benchtime", "run each benchmark for duration `d`")
}

var (
	matchBenchmarks *string
	benchmarkMemory *bool

	benchTime = durationOrCountFlag{d: 1 * time.Second} // changed during test of testing package
)
```

Reply:
Updated.